src/deepsparse/v2/operators/operator.py

```diff
@@ -87,4 +89,36 @@ def __call__(
             inference_input = self.input_schema(**kwargs)
         else:
             inference_input = kwargs
-        return self.run(inference_input, context=context)
+
+        run_output = self.run(inp=inference_input, context=context)
+
+        if self.has_output_schema():
+            return self.output_schema(**run_output)
+        return run_output
+
+    @abstractmethod
+    def run(self, inp: Any, context: Optional[Context]) -> Any:
+        """
+        :param inp: operator input, as the defined input schema if applicable
+        :param context: pipeline context of already run operators
+        :return: result of this operator as the defined output schema if applicable
+        """
+        raise NotImplementedError
+
+    def expand_inputs(self, **kwargs):
+        """
+        Generic function to handle expanding values.
+        """
+        raise NotImplementedError
+
+    def condense_inputs(self, **kwargs):
```

Comment:
what's the reasoning for making this a top level function? are routers always going to call condense/expand?

Reply:
keeping for now, will be updated in a following PR once I add in split/joining on the pipeline level